src/Startup/Hosting/OwinHost.cs

```diff
@@ -0,0 +1,75 @@
+        public static IWebHostBuilder UseOwinServer(this IWebHostBuilder builder) =>
+            builder.ConfigureServices(services => services.AddSingleton<IServer, OwinServer>());
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="WebHostBuilder"/> class with pre-configured defaults.
+        /// </summary>
+        /// <remarks>
+        ///   The following defaults are applied to the returned <see cref="WebHostBuilder"/>:
+        ///     use OWIN as the web server,
+        ///     use the default <see cref="IHostingEnvironment.ContentRootPath"/> (typically <see cref="AppContext.BaseDirectory"/>)
+        ///     load <see cref="IConfiguration"/> from 'appsettings.json' and 'appsettings.[<see cref="IHostingEnvironment.EnvironmentName"/>].json',
+        ///     load <see cref="IConfiguration"/> from User Secrets when <see cref="IHostingEnvironment.EnvironmentName"/> is 'Development' using the entry assembly,
+        ///     load <see cref="IConfiguration"/> from environment variables,
+        ///     configures the <see cref="ILoggerFactory"/> to log to debug but not console,
+        ///     and enables the ability for frameworks to bind their options to their default configuration sections.
+        /// </remarks>
+        /// <returns>The initialized <see cref="IWebHostBuilder"/>.</returns>
+        public static IWebHostBuilder CreateDefaultBuilder()
+        {
+            // https://github.com/dotnet/aspnetcore/blob/c2cfc5f140cd2743ecc33eeeb49c5a2dd35b017f/src/DefaultBuilder/src/WebHost.cs#L150
+            var builder = new WebHostBuilder()
+                .UseOwinServer()
+                .ConfigureAppConfiguration((hostingContext, config) =>
+                {
+                    var env = hostingContext.HostingEnvironment;
+
+                    config.AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                          .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true, reloadOnChange: true);
+
+                    if (env.IsDevelopment())
+                    {
+                        var appAssembly = Assembly.Load(new AssemblyName(env.ApplicationName));
+                        if (appAssembly != null)
+                        {
+                            config.AddUserSecrets(appAssembly, optional: true);
+                        }
+                    }
+
+                    config.AddEnvironmentVariables();
+                })
+                .ConfigureLogging((hostingContext, logging) =>
+                {
+                    logging.AddConfiguration(hostingContext.Configuration.GetSection("Logging"));
+                    logging.AddDebug();
+                })
+                // Skipping PostConfigure<HostFilteringOptions>()
+                // Skipping ConfigurationChangeTokenSource<HostFilteringOptions>
+                // Skipping internal HostFilteringStartupFilter
+                // Skipping UseIISIntegration()
```

Comment:
@davidfowl I had a hard time telling if/how [these parts of `WebHost.CreateDefaultBuilder()`](https://github.com/dotnet/aspnetcore/blob/c2cfc5f140cd2743ecc33eeeb49c5a2dd35b017f/src/DefaultBuilder/src/WebHost.cs#L187-L206) would impact a project hosted outside of Kestrel.